lib/streamaggr/streamaggr.go

```diff
@@ -172,11 +208,71 @@ func (a *Aggregators) Push(tss []prompbmarshal.TimeSeries) {
 	if a == nil {
 		return
 	}
+
+	a.mu.RLock()
+	defer a.mu.RUnlock()
+
 	for _, aggr := range a.as {
 		aggr.Push(tss)
 	}
 }
 
+// ReInitConfigs reinits state of Aggregators a with the given new stream aggregation config
+func (a *Aggregators) ReInitConfigs(cfgs []*Config) error {
```

Comment:
would you mind covering this func with tests?

Reply:
done